packages/core-utils/src/watcher.ts

```diff
@@ -81,18 +81,17 @@ export class Watcher {
       const startingBlock = Math.max(blockNumber - this.NUM_BLOCKS_TO_FETCH, 0)
       const successFilter: ethers.providers.Filter = {
         address: layer.messengerAddress,
-        topics: [ethers.utils.id(`RelayedMessage(bytes32)`)],
-        fromBlock: startingBlock
+        topics: [ethers.utils.id(`RelayedMessage(bytes32)`), msgHash],
```

Comment:
For other reviewers, this is indeed the correct way to use indexed parameters in a filter. See these Ethers docs for more information: https://docs.ethers.io/v5/concepts/events/. Example from those docs:

```
filter = {
    address: tokenAddress,
    topics: [
        id("Transfer(address,address,uint256)"),
        hexZeroPad(myAddress, 32)
    ]
};

(We don't need `hexZeroPad` because `msgHash` is already 32 bytes.
```

Reply:
Without utilizing the `Contract` object itself, this looks fine to me. The `contract.filters.RelayedMessage(msgHash)` syntax is nice for generating the `ethers.providers.Filter` object. I think such a change is out of scope of this PR as its a larger change

Reply:
Thanks for the feedback both!

I wasn't sure if including `hexZeroPad()` by default is a good defensive posture in case the event type changes. It seems like you wouldn't want the event definition to change over time as that would complicate querying the logs. In that case you wouldn't need `hexZeroPad()` as a defensive mechanism. I also noticed that previously we were using a strict equality operator. This made me think that directly filtering on `msgHash` was fine.
```
matches = logs.filter((log: ethers.providers.Log) => log.data === msgHash)
```